lib/SimpleSAML/Database.php

```diff
@@ -137,14 +138,14 @@ private static function generateInstanceId($config)
      * This function connects to a database.
      *
      * @param string $dsn Database connection string
-     * @param string $username SQL user
-     * @param string $password SQL password
+     * @param string|null $username SQL user
+     * @param string|null $password SQL password
      * @param array  $options PDO options
      *
      * @throws \Exception If an error happens while trying to connect to the database.
      * @return \PDO object
      */
-    private function connect($dsn, $username, $password, $options)
+    private function connect(string $dsn, $username, $password, array $options): PDO
```

Comment:
Ideally I'd typehint them as `?string`, but that's PHP 7.1..
They are already given the default value of `null` on line 90 of this file, so it seemed overkill to do `string $username = null`..  I figured to leave them for now and properly typehint them in 2.0..

Reply:
Yeah, that was my first reaction too.

But I guess we can still benefit from typehints there now, right? Otherwise we still need to check them manually...